source/non_matching/mesh_classifier.cc

```diff
@@ -333,6 +334,9 @@ namespace NonMatching
                  "The elements in the FECollection of the incoming DoFHandler "
                  "must have face support points."));
       }
+#else
+    Assert(false, ExcNeedsLAPACK());
+#endif
```

Comment:
```suggestion
#ifndef DEAL_II_WITH_LAPACK
    AssertThrow(false, ExcNeedsLAPACK());
#endif
```
The upper part of the conditional doesn't need LAPACK, right?

Reply:
Yes, but the class doesn't make sense without lapack anyway so we might as well guard it.

I'd prefer to keep it in the original version because this is how we typically handle constructors with missing dependencies (e.g., MPI).

Reply:
using `AssertThrow` is right - I'll change that.